waffle_hub/utils/data.py

```diff
@@ -71,35 +71,35 @@ def resize_image(
     W, H = image_size
 
     if letter_box:
-        if w > h:
-            ratio = W / w
-            w_ = W
-            h_ = round(h * ratio)
+        h_ratio = H / h
+        w_ratio = W / w
 
-            total_pad = w_ - h_
+        if w_ratio < h_ratio:
+            resize_shape = (int(w*w_ratio), round(h*w_ratio))
+            total_pad = H - resize_shape[1]
             top = total_pad // 2
             bottom = total_pad - top
             left, right = 0, 0
-
+            w_ = resize_shape[0]
+            h_ = resize_shape[1] + total_pad
         else:
-            ratio = H / h
-            w_ = round(w * ratio)
-            h_ = H
-
-            total_pad = h_ - w_
+            resize_shape = (round(w*h_ratio), int(h*h_ratio))
+            total_pad = W - resize_shape[0]
             left = total_pad // 2
             right = total_pad - left
             top, bottom = 0, 0
+            w_ = resize_shape[0] + total_pad
+            h_ = resize_shape[1]
```

Comment:
Aren't they same with W, H?
`new_shape`(w_, h_) was intended to be stand for the resized image size before pad.

Reply:
I'll match it.